docker-api/src/main/java/com/yahoo/vespa/hosted/dockerapi/CreateContainerCommandImpl.java

```diff
@@ -167,34 +167,34 @@ private CreateContainerCmd createCreateContainerCmd() {
         List<Bind> volumeBinds = volumeBindSpecs.stream().map(Bind::parse).collect(Collectors.toList());
 
         final HostConfig hostConfig = new HostConfig()
-                .withSecurityOpts(new ArrayList<>(securityOpts));
+                .withSecurityOpts(new ArrayList<>(securityOpts))
+                .withBinds(volumeBinds)
+                .withUlimits(ulimits)
+                .withCapAdd(addCapabilities.toArray(new Capability[0]))
+                .withCapDrop(dropCapabilities.toArray(new Capability[0]))
+                .withPrivileged(privileged);
 
         containerResources.ifPresent(cr -> hostConfig
                 .withCpuShares(cr.cpuShares())
                 .withMemory(cr.memoryBytes())
                 // MemorySwap is the total amount of memory and swap, if MemorySwap == Memory, then container has no access swap
                 .withMemorySwap(cr.memoryBytes())
-                .withCpuPeriod(cr.cpuQuota() > 0 ? cr.cpuPeriod() : null)
-                .withCpuQuota(cr.cpuQuota() > 0 ? cr.cpuQuota() : null));
+                .withCpuPeriod(cr.cpuQuota() > 0 ? (long) cr.cpuPeriod() : null)
```

Comment:
an cpuPeriod() and cpuQuota() be updated to return long?

Reply:
Not really, for some reason the update command requires `Integer`: https://github.com/docker-java/docker-java/blob/c5c89327108abdc50aa41d7711d4743b8bac75fe/src/main/java/com/github/dockerjava/api/command/UpdateContainerCmd.java#L25:L32